pkg/workflows/upgrade.go

```diff
@@ -188,14 +188,20 @@ func (s *upgradeCoreComponents) Run(ctx context.Context, commandContext *task.Co
 	}
 	commandContext.UpgradeChangeDiff.Append(changeDiff)
 
-	changeDiff, err = commandContext.AddonManager.Upgrade(ctx, target, commandContext.CurrentClusterSpec, commandContext.ClusterSpec)
+	changeDiff, err = commandContext.ClusterManager.Upgrade(ctx, target, commandContext.CurrentClusterSpec, commandContext.ClusterSpec)
 	if err != nil {
 		commandContext.SetError(err)
 		return &CollectDiagnosticsTask{}
 	}
 	commandContext.UpgradeChangeDiff.Append(changeDiff)
 
-	changeDiff, err = commandContext.ClusterManager.Upgrade(ctx, target, commandContext.CurrentClusterSpec, commandContext.ClusterSpec)
+	err = commandContext.AddonManager.UpdateLegacyFileStructure(ctx, commandContext.ClusterSpec)
+	if err != nil {
+		commandContext.SetError(err)
+		return nil
```

Comment:
consider if we want to return the new diagnostics task here, which executes the support bundle on the workload and mgmt clusters